### DISCUSSION TITLE
Is there support for eventlet workers ?

### DISCUSSION THREAD

**Discussion by rm2631** [openai/openai-python#1202] [Q&A]
I'm currently migrating from 0.x to 1.7.2 and having issues after deploying to my staging environment. 

- At first, whenever I instantiated an AzureOpenAI object, I would get this error: "maximum recursion depth exceeded while calling a Python object".
- I isolated the client instantiation in another endpoint at it's most basic and still got the issue.
- I was able to trace it back to the worker. Switching to a synchronous worker works as intended. 
- Using the AsyncAzureOpenAI yields the same failures.

I'm currently using eventlet because the server also hosts a socket-io connection.

So, my question is, is there support that I'm missing on eventlet workers ?  

Cheers.

What my implementation looks like : 

```
params = {
        "temperature": 0,
        "model": "model-aiservicesic-gpt-35-turbo",
        "messages": [*history, {"role": "user", "content": input}],
    }
client = AsyncAzureOpenAI(
    azure_ad_token_provider=get_bearer_token_provider(
        credentials, "https://cognitiveservices.azure.com/.default"
    ),
    api_version="2023-05-15",
    azure_endpoint=os.environ.get("ic_openAIEndpoint"),
)
```

**Comment by rattrayalex** [comment]
Can you share a fully working example that reproduces the issue?
